src/components/Tabs/ScatterPlotTab.tsx

```diff
@@ -142,26 +139,56 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
     dataset !== props.dataset || colorRampMin !== props.colorRampMin || colorRampMax !== props.colorRampMax;
 
   /**
-   * Wrapper around useState that signals the render spinner whenever the values are set, and
-   * uses useTransition to deprioritize the state update.
+   * Wrapper around a prop value and its setter. Triggers a callback whenever the values are set or the
+   * prop value changes, and tracks its own copy of the prop value as state to defer re-renders.
+   * @param propValue The prop value to track.
+   * @param setPropValue The setter for the prop value.
+   * @param onChange An optional callback to call when the prop value changes.
+   * @returns A tuple with the current value and a setter that triggers a render.
    */
-  const useRenderTriggeringState = <T extends any>(initialValue: T): [T, (value: T) => void] => {
-    const [value, _setValue] = useState(initialValue);
-    const setState = (newValue: T): void => {
-      if (newValue === value) {
+  const usePropTransitionWrapper = <T extends any>(
+    propValue: T,
+    setPropValue: (newValue: T) => void,
+    onChange?: () => void
+  ): [T, (value: T) => void] => {
+    const [value, _setValue] = useState(propValue);
+
+    // Returned setter which will flag that a render is occurring.
+    const setState = (newPropValue: T): void => {
+      if (newPropValue === propValue) {
         return;
       }
-      setIsRendering(true);
-      startTransition(() => {
-        _setValue(newValue);
-      });
+      setPropValue(newPropValue);
+      onChange && onChange();
     };
+
+    // Trigger a render when the prop value changes.
+    useEffect(() => {
+      if (propValue !== value) {
+        onChange && onChange();
+        startTransition(() => {
+          _setValue(propValue);
+        });
+      }
+    }, [propValue]);
     return [value, setState];
   };
```

Comment:
Would like a gut check here on if I made this too complicated, or if I could make the comments clearer.